src/main/java/io/confluent/kafkarest/SimpleConsumerPool.java

```diff
@@ -81,22 +237,43 @@ synchronized public SimpleFetcher get(final String host, final int port) {
         throw Errors.simpleConsumerPoolTimeoutException();
       }
     }
-
   }
 
-  synchronized public void release(SimpleFetcher simpleFetcher) {
-    log.debug("Releasing into the pool SimpleConsumer with id " + simpleFetcher.clientId());
-    availableConsumers.add(simpleFetcher.clientId());
+  synchronized public void release(Consumer<byte[], byte[]> consumer, TopicPartition topicPartition, long offset) {
+    Map<Long, List<Consumer<byte[], byte[]>>> offsetsMap =
+      topicPartitionOffsetConsumers.get(topicPartition);
+    if (offsetsMap != null) {
```

Comment:
This seems more complicated than it needs to be. Can't we just add a `new LinkedList<>()` to `topicPartitionOffsetConsumers` if it isn't there, then do the same for `offsetsMap` and `consumers`, then finally do a single `consumers.add(consumer)`? Here it seems we're repeating a number of steps by breaking down every case into its own block.